src/Kiota.Builder/Extensions/CodeParametersEnumerableExtensions.cs

```diff
@@ -1,15 +1,13 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Linq;
 
 using Kiota.Builder.CodeDOM;
 
 namespace Kiota.Builder.Extensions;
 public static class CodeParametersEnumerableExtensions {
-    public static CodeParameter OfKind(this IEnumerable<CodeParameter> parameters, CodeParameterKind kind) {
+    public static CodeParameter? OfKind(this IEnumerable<CodeParameter> parameters, CodeParameterKind kind) {
+        ArgumentNullException.ThrowIfNull(parameters);
```

Comment:
Should we also filter for null members? i.e `parameters.Filter(p => p != null)`?
See [Nullable Reference: Known Pitfalls](https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references#known-pitfalls)

Reply:
added